File: py/Climbing_Stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 1:
            return 1
        dp = [1] * 2
        for i in range(2, n + 1):
            dp[1], dp[0] = dp[1] + dp[0], dp[1]
        return dp[1]

if __name__ == '__main__':
    s = Solution()
    print(s.climbStairs(5))

File: c_plus/Climbing_Stairs.cpp
#include <iostream>
using namespace std;

class Solution {
public:
    int climbStairs(int n) {
        if (n <= 1) return 1;
        int dp[2] = {1, 1};
        for (int i = 2; i < n + 1; i++) {
            int temp = dp[1];
            dp[1] = dp[1] + dp[0];
            dp[0] = temp;
        }
        return dp[1];
    }
};

int main() {
    Solution s;
    cout << s.climbStairs(4) << endl;
}